### PR TITLE
Adding ansible role for SonarQube

### PR DESCRIPTION
**Pull request by Pranshu021**
Adding ansible role for SonarQube

Playbooks: 
- oah_install.yml
- oah_update.yml
- oah_remove.yml
- oah_validate.yml

@asa1997 Please review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,38 @@
-# ansible-role-bes-sonarQube
-Ansible role for SonarQube
+Role Name
```
Comment (asa1997) [inline README.md:1]: Please add some content

File: tasks/oah-install.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+
+- name: Running Java version command
+  shell: "java --version"
+  register: java_output
+
+- name: Check if Java is installed
```
Comment (asa1997) [inline tasks/oah-install.yml]: syntax error here

image

**Review by asa1997 — CHANGES_REQUESTED** [review]
@Pranshu021 Please address the comments

**Comment by asa1997** [comment]
@Pranshu021 Please also provide a meaningful title to the pull request.

File: tasks/oah-install.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+
+- name: Running Java version command
+  shell: "java --version"
+  register: java_output
+
+- name: Check if Java is installed
+  failed_when: '"Error" in java_output.stdout'
+
+- name: Checking Latest version of SonarQube.
```
Comment (asa1997) [inline tasks/oah-install.yml:13]: ERROR! conflicting action statements: shell, warn

image

File: tasks/oah-update.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+
+- name: Checking Latest version of SonarQube.
```
Comment (asa1997) [inline tasks/oah-update.yml:3]: ERROR! conflicting action statements: shell, warn
image

File: tasks/oah-install.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+- name: Checking Latest version of SonarQube.
+  shell: 'sudo curl -s https://api.github.com/repos/sonarsource/sonarqube/releases/latest | grep "tarball_url" | cut -d : -f 2,3 | tr -d \" | tr -d "," | xargs | grep -Eo "([0-9]|\.)*(?=\s|$)"'
+  register: sonar_latest_version
+  warn: false
+
+- name: Downloading and Extracting SonarQube
+  unarchive:
+    src: "{{ sonar_download_url }}/sonarqube-{{sonar_latest_version.stdout}}.zip"
+    dest: /opt/
```
Comment (asa1997) [inline tasks/oah-install.yml:21]: need root permission for this operation

File: tasks/oah-remove.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+---
+
+- name: Checking the installed version
+  shell: 'ls /opt/ | grep sonar | grep -Eo "([0-9]|\.)*(?=\s|$)"'
+  register: sonar_installed_version
+
+- name: Uninstall Sonarqube
```
Comment (asa1997) [inline tasks/oah-remove.yml:7]: Require sudo permission

File: tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+---
+# tasks file for ansible-role-bes-sonarQube
+
+- include: "tasks/oah-install.yml"
+  when: oah_command == "install"
+
+- include: "tasks/oah-remove.yml"
+  when: oah_command == "uninstall"
```
Comment (asa1997) [inline tasks/main.yml]: command should be remove

File: tasks/oah-validate.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+---
+
+- name: Checking the installed version
```
Comment (asa1997) [inline tasks/oah-validate.yml]: Fails when sonarqube is not found. 
image

Suggesting to add some error handling.

File: tasks/oah-install.yml (hunk 3 above)
Comment (Pranshu021) [reply tasks/oah-install.yml:13]: Removed the standalone failed_when task and changed the validating condition. Now working

File: tasks/oah-update.yml (hunk 4 above)
Comment (Pranshu021) [reply tasks/oah-update.yml:3]: Removed the conflicting statement warn which had no effect in functionality. Working now

File: tasks/oah-install.yml (hunk 5 above)
Comment (Pranshu021) [reply tasks/oah-install.yml:21]: Added the required statement

File: tasks/oah-remove.yml (hunk 6 above)
Comment (Pranshu021) [reply tasks/oah-remove.yml:7]: Added the required statement

File: tasks/main.yml (hunk 7 above)
Comment (Pranshu021) [reply tasks/main.yml]: Change is made already [ uninstall -> remove ]

File: tasks/oah-validate.yml (hunk 8 above)
Comment (Pranshu021) [reply tasks/oah-validate.yml]: Improved the functioning of oah-validate.
Introduced error handling

**Comment by Pranshu021** [comment]
@asa1997 Made all the required changes. Please check

**Review by asa1997 — APPROVED** [review]
@Pranshu021 Please address the comments.
- Please do add the required contents in the readme file
- Do add the code for reset
@panickervinod @Pranshu021 LGTM.

**Comment by Pranshu021** [comment]
@panickervinod @sumodgeorge  Added the Reset file and README.md too. Please check @asa1997

**Pull request merged**